model/src/main/proto/survey.proto

```diff
@@ -0,0 +1,204 @@
+    SurveyQuestionOptionList multiple_choice_with_other = 5;
+
+    // Indicates whether this question takes a free-text response.
+    bool free_form_text = 6;
+  }
+
+  // The language code to which this question is localized.
+  string language_code = 7;
+}
+
+// Represents a list of answer options for a multiple choice question.
+message SurveyQuestionOptionList {
+  repeated SurveyQuestionOption options = 1;
+}
+
+// Corresponds to the semantic name given to each question.
+enum QuestionName {
+  // The question name is unknown.
+  QUESTION_NAME_UNSPECIFIED = 0;
+
+  // Corresponds to the name of the user type question.
+  USER_TYPE = 1;
+
+  //  Corresponds to the market fit question.
+  MARKET_FIT = 2;
+
+  // Corresponds to the NPS question.
+  NPS = 3;
+
+  // Corresponds to the open-ended question for a NPS promoter.
+  PROMOTER_FEEDBACK = 4;
+
+  // Corresponds to the open-ended question for a NPS passive.
+  PASSIVE_FEEDBACK = 5;
+
+  // Corresponds to the open-ended question for a NPS detractor.
+  DETRACTOR_FEEDBACK = 6;
+}
+
+// Represents the multiple choice answer available for a question.
+message SurveyQuestionOption {
+  // The ID of this question option. It is a string that uniquely identifies the question option
+  // within the context of a survey question. It is only unique within the scope of a survey.
+  string question_option_id = 1;
+
+  // The semantic name of the question this answer option is tied to.
+  QuestionName question_name = 2;
+
+  // The actual text value of the answer option.
+  string label = 3;
+
+  // The type of answer this option represents.
+  oneof answer {
+    // The enum representation of the type of user answer option.
+    UserTypeAnswer user_type = 4;
+
+    // The enum representation of the market fit answer option.
+    MarketFitAnswer market_fit = 5;
+
+    // The integer value representing the nps score option.
+    int32 nps = 6;
+  }
+}
+
+// Represents an answer that was selected by a user for a survey question.
+message SurveyUserAnswer {
+  // The ID of the survey to which this answer is tied. This is a string value that uniquely
+  // identifies the survey globally.
+  string survey_id = 1;
+
+  // Semantic name of the question the response is tied to.
+  QuestionName question_name = 2;
+
+  // The value of the answer the user gave.
+  oneof answer {
+    // The enum representation of the type of user answer selected.
+    UserTypeAnswer user_type = 3;
+
+    // The enum representation of the market fit answer selected.
+    MarketFitAnswer market_fit = 4;
+
+    // The integer value representing the nps score selected by the user.
+    int32 nps = 5;
+  }
+}
+
+// Represents the different types of users who could be taking the survey.
+// Consists of answer options for the user type question.
+enum  UserTypeAnswer {
+  // User type unknown.
+  USER_TYPE_UNSPECIFIED = 0;
+
+  // Corresponds to an unspecified or unknown type of user.
```

Comment:
Out of curiosity, why not use ``USER_TYPE_UNSPECIFIED`` for this case? Why have a different OTHER enum?

Reply:
Loosely speaking, it would be more difficult to distinguish between responses that were genuinely unspecified and those that were intended to be "other" choices. However the implementation should have checks to ensure each question has a selected answer before proceeding to the next question.